test/common/network/connection_impl_test.cc

```diff
@@ -472,6 +474,23 @@ TEST_P(ConnectionImplTest, WatermarkFuzzing) {
   disconnect(true);
 }
 
+TEST_P(ConnectionImplTest, BindTest) {
+  std::string address_string = "127.0.0.9";
+  if (GetParam() == Network::Address::IpVersion::v4) {
+    source_address_ = Network::Address::InstanceConstSharedPtr{
+        new Network::Address::Ipv4Instance(address_string, 0)};
+  } else {
+    address_string = "::1";
```

Comment:
This test probably doesn't validate the new behavior for IPv6, since we were probably already binding here, but I don't have a better solution.